pkg/scheduler/factory/factory.go

```diff
@@ -1055,6 +1058,116 @@ func (c *configFactory) invalidateCachedPredicatesOnNodeUpdate(newNode *v1.Node,
 	}
 }
 
+func nodeSchedulingPropertiesChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	if nodeAllocatableChanged(newNode, oldNode) {
+		glog.V(4).Infof("Allocatable resource of node %s changed", newNode.Name)
+		return true
+	}
+	if nodeLabelsChanged(newNode, oldNode) {
+		glog.V(4).Infof("Labels of node %s changed", newNode.Name)
+		return true
+	}
+	if nodeTaintsChanged(newNode, oldNode) {
+		glog.V(4).Infof("Taints of node %s changed", newNode.Name)
+		return true
+	}
+	if nodeConditionsChanged(newNode, oldNode) {
+		glog.V(4).Infof("Conditions of node %s changed", newNode.Name)
+		return true
+	}
+	if newNode.Spec.Unschedulable != oldNode.Spec.Unschedulable && newNode.Spec.Unschedulable == false {
+		glog.V(4).Infof("Node %s changed to schedulable", newNode.Name)
+		return true
+	}
+	return false
+}
+
+func nodeAllocatableChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	oldAllocatable := oldNode.Status.Allocatable
+	newAllocatable := newNode.Status.Allocatable
+	if reflect.DeepEqual(oldAllocatable, newAllocatable) {
+		return false
+	}
+	for resource, newValue := range newAllocatable {
+		// If new node contains an allocatable resource which not exists in old node, or
+		// according value is bigger than that in old node, we assume this node becomes more schedulable.
+		oldValue, exist := oldAllocatable[resource]
+		if !exist || newValue.Cmp(oldValue) > 0 {
+			return true
+		}
+	}
+	return false
+}
+
+func nodeLabelsChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	return !reflect.DeepEqual(oldNode.GetLabels(), newNode.GetLabels())
+}
+
+func nodeTaintsChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	if !reflect.DeepEqual(newNode.Spec.Taints, oldNode.Spec.Taints) {
+		return true
+	}
+	oldTaints, oldErr := helper.GetTaintsFromNodeAnnotations(oldNode.GetAnnotations())
+	if oldErr != nil {
+		// If parse old node's taint annotation failed, we assume node's taint changed.
+		glog.Errorf("Failed to get taints from annotation of old node %s: %v", oldNode.Name, oldErr)
+		return true
+	}
+	newTaints, newErr := helper.GetTaintsFromNodeAnnotations(newNode.GetAnnotations())
+	if newErr != nil {
+		// If parse new node's taint annotation failed, we assume node's taint changed.
+		glog.Errorf("Failed to get taints from annotation of new node %s: %v", newNode.Name, newErr)
+		return true
+	}
+	if !reflect.DeepEqual(oldTaints, newTaints) {
+		return true
+	}
+	return false
+}
+
+func nodeConditionsChanged(newNode *v1.Node, oldNode *v1.Node) bool {
+	if reflect.DeepEqual(oldNode.Status.Conditions, newNode.Status.Conditions) {
+		return false
+	}
+
+	healthyConditions := []v1.NodeConditionType{v1.NodeReady}
```

Comment:
Similar to my previous comment, I would return true as long as old and new conditions are not equal. This will help reduce chances of introducing bugs in the future when new node conditions are added.

Reply:
Done